### PR TITLE
Upgrade OpenSSH to 9.8p1 to resolve the Regresshion CVE

### PR DESCRIPTION
**Pull request by jslobodzian**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] If you are adding/removing a .spec file that has multiple-versions supported, please add [@microsoft/cbl-mariner-multi-package-reviewers](https://github.com/orgs/microsoft/teams/cbl-mariner-multi-package-reviewers) team as reviewer [(Eg. golang has 2 versions 1.18, 1.21+)](https://github.com/microsoft/azurelinux/tree/2.0/SPECS/golang)
- [x] Ready to merge

---

###### Summary 
This upgrade resolves Critical CVE-2024-6387 which is a regression for CVE-2006-5051.  Note that this version of openssh splits the server into two binaries, so there is a packaging difference from the previous version.  The server/listener (sshd) and a separate binary that hosts the session (sshd-session). 

###### Does this affect the toolchain?  

**NO**

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2024-6387

###### Test Methodology

Buddy build and manually installed the client and server package.  Verified that I could connect to the server and verified that the ssh client functioned.

Note that the pTest currently fails for ssh because the Azure Linux version of openssh does not include the secure-key (libraries).  A separate bug is tracking to resolve post GA.

### CODE REVIEW

**Review by nisamson — APPROVED** [review]

File: SPECS/openssh/openssh.spec [hunk 1]
```diff
@@ -261,6 +262,9 @@ fi
 %{_mandir}/man8/ssh-sk-helper.8.gz
 
 %changelog
+* Mon Jul 01 2024 Jon Slobodzian <joslobo@microsoft.com - 9.8p1-1
```
Comment (anphel31) [inline SPECS/openssh/openssh.spec:265]: typo  <joslobo@microsoft.com

File: SPECS/openssh/openssh.spec [hunk 2]
```diff
@@ -1,4 +1,4 @@
-%global openssh_ver 9.7p1
+\%global openssh_ver 9.8p1
```
Comment (anphel31) [inline SPECS/openssh/openssh.spec:1]: typo? (backslash)

File: SPECS/openssh/openssh.spec (hunk 1 above)
Comment (jslobodzian) [reply SPECS/openssh/openssh.spec]: Fixed!

File: SPECS/openssh/openssh.spec [hunk 3]
```diff
@@ -261,6 +262,9 @@ fi
 %{_mandir}/man8/ssh-sk-helper.8.gz
 
 %changelog
+* Mon Jul 01 2024 Jon Slobodzian <joslobo@microsoft.com> - 9.8p1-1
+* Upgrade to version 9.8p1. This fixes CVE-2024-6387 (a regression to CVE-2006-5051) in OpenSSH's server.
```
Comment (jslobodzian) [inline SPECS/openssh/openssh.spec:266]: should be a dash here

Comment (jslobodzian) [reply SPECS/openssh/openssh.spec]: fixed!

File: SPECS/openssh/openssh.spec (hunk 2 above)
Comment (jslobodzian) [reply SPECS/openssh/openssh.spec]: Fixed!

**Review by anphel31 — APPROVED** [review]

**Comment by mfrw** [comment]
Fixes:
- #9555

**Pull request merged**